### PR TITLE
Add pandera third-party tests

### PR DESCRIPTION
**Pull request by Viicos**
## Change Summary

Part of https://github.com/pydantic/pydantic/issues/11160.

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: .github/workflows/third-party.yml [hunk 1]
```diff
@@ -214,6 +220,50 @@ jobs:
...
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'schedule' && github.repository == 'pydantic/pydantic') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'third-party-tests'))
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11']
```
Comment (sydney-runkle) [inline .github/workflows/third-party.yml:235]: No 3.12?

File: .github/workflows/third-party.yml [hunk 2]
```diff
@@ -2,6 +2,12 @@
 # https://github.com/python/typing_extensions/blob/main/.github/workflows/third_party.yml.
 # Running the tests of various third-party libraries that use Pydantic. This helps us spot regressions early, and helps
 # flag when third-party libraries are making incorrect assumptions that might cause them to break when we cut a new release.
+# Instructions:
+# - The CI should be as close as possible to the original project's workflows.
+# - Use the checkout action to clone Pydantic with a custom path (e.g. `path: pydantic-latest`).
+# - Make sure Pydantic is installed in editable mode (e.g. `uv pip install -e ./pydantic-latest`)
+#   so that the path appears in the `pip list` output (and so we can be assured Pydantic was properly
+#   installed from the provided path).
```
Comment (sydney-runkle) [inline .github/workflows/third-party.yml:10]: Good to add these here. There's so much redundancy across jobs, maybe we could consolidate with shared logic for installing pydantic and settings things up...

**Review by sydney-runkle — COMMENTED** [review]
LGTM, just some questions about our general structure for this test file.

File: .github/workflows/third-party.yml (hunk 1 above)
Comment (Viicos) [reply .github/workflows/third-party.yml:235]: Seems like they are lagging behind Python versions..

**Review by sydney-runkle — APPROVED** [review]
Nice, thanks!

**Pull request merged**